### PR TITLE
fix(bulk-import): bulk import ux improvements

### PR DESCRIPTION
**Pull request by debsmita1**
Resolves:
- https://issues.redhat.com/browse/RHIDP-3792
- https://issues.redhat.com/browse/RHIDP-3607
- https://issues.redhat.com/browse/RHIDP-3798
- https://issues.redhat.com/browse/RHIDP-3783
- https://issues.redhat.com/browse/RHIDP-3837
- https://issues.redhat.com/browse/RHIDP-3858
- https://issues.redhat.com/browse/RHIDP-3888
- https://issues.redhat.com/browse/RHIDP-3893

**Fixes:**

- Removed the `Repositories` tab view and simply listing the added repositories
- Moved the `DeleteDialogContext` to `shared-react` package. Updated the rbac UI plugin
- In the `Add Repositories` list page show the PR contents in the sidepanel if the repo has a PR waiting for approval
- Text on the educational card is now centered
- The import PR description now links to `<base-url>/catalog` and the `<base-url>/bulk-import/repositories?repository=<repo-url>` which opens the sidepanel for the repository with the Pull request details on the app
- Fixed the dark theme issue
- updating the selected rows in the `Added repositories` list now triggers the API with the updated size per integration param
- Fixed the issue with the drawer closing when the added repository list updates
- Added new tests

**Screenshots:**
Showing an info panel with the PR link if the job is waiting for approval
Screenshot 2024-09-09 at 1 23 05 PM

Tabbed layout in the `Added repositories` list page is removed
Screenshot 2024-09-03 at 11 24 00 PM

Text on the educational card is now centered
Screenshot 2024-09-03 at 11 23 41 PM

The import PR description links to Software catalog `<base-url>/catalog` & associated import job `<base-url>/bulk-import/repositories?repository=<repo-url>`

Screenshot 2024-09-06 at 12 36 59 AM

Screenshot 2024-09-06 at 12 37 11 AM

Shows an alert instead of 404 error page when the user is not authorized to access the plugin (https://issues.redhat.com/browse/RHIDP-3833)
Screenshot 2024-09-04 at 3 20 30 PM

Screenshot 2024-09-10 at 12 56 02 PM

Add repositories page in dark theme on showcase app

Screenshot 2024-09-04 at 10 06 47 PM

Screenshot 2024-09-04 at 10 07 10 PM

Screenshot 2024-09-09 at 3 54 59 PM

### CODE REVIEW

**Review by rm3l — COMMENTED** [review]
Thanks for these improvements. I quickly tested this and it looks good overall (especially with the dark theme); But I noticed the following behavior:

1. When selecting more than one repository from the organizations tab, I noticed that the form with the PR details is only filled for the first repo in the list. See the screenshots below. Is that expected?

image

image

2. Does this PR really fix https://issues.redhat.com/browse/RHIDP-3798 (as depicted in this PR description)? It seems to me that @jerolimov's point was to make the link open the bulk import details for the selected repo, so they can edit the PR directly. But right now, the link points to the current instance Backstage catalog (which was the point of https://issues.redhat.com/browse/RHIDP-3792):

image

**Comment by debsmita1** [comment]
> Thanks for these improvements. I quickly tested this and it looks good overall (especially with the dark theme); But I noticed the following behavior:
> 
> 1. When selecting more than one repository from the organizations tab, I noticed that the form with the PR details is only filled for the first repo in the list. See the screenshots below. Is that expected?
> 
> image
> 
> image
> 

Right, this is the issue that I am trying to fix which is why I made the PR WIP


> 2. Does this PR really fix https://issues.redhat.com/browse/RHIDP-3798 (as depicted in this PR description)? It seems to me that @jerolimov's point was to make the link open the bulk import details for the selected repo, so they can edit the PR directly. But right now, the link points to the current instance Backstage catalog (which was the point of https://issues.redhat.com/browse/RHIDP-3792):
> 
> image

Oh I missed that, on it right away

**Comment by ShiranHi** [comment]
Thank you for all these @debsmita1 !

Some quesitons:

> Showing an info panel with the PR link if the job is waiting for approval
Can we make this message to match the [design](https://www.figma.com/design/09Iltd57RiuqOtKWkjy8MH/Bulk-import?node-id=1061-1660&t=NgIpoP1C1Jx3jhyk-1)? without the description text?

> Shows an alert instead of 404 error page when the user is not authorized to access the plugin (https://issues.redhat.com/browse/RHIDP-3833)
I guess this is fine for now, but it will be great using the [full page layout] (https://www.figma.com/design/GeVb7rc390TkxeQNnJIcBD/RHDH-Theme-(pages)?node-id=110-933&t=tedPPRSTpsge64S3-1) for this use cases in the future (probably we need another ticket).

>  Add repositories page in dark theme on showcase app
The text on the educational card in this screenshot doesn’t seem to line up with the illustrations

**Comment by debsmita1** [comment]
> Thank you for all these @debsmita1 !
> 
> Some questions:
> 
> > Showing an info panel with the PR link if the job is waiting for approval
> 
> Can we make this message to match the [design](https://www.figma.com/design/09Iltd57RiuqOtKWkjy8MH/Bulk-import?node-id=1061-1660&t=NgIpoP1C1Jx3jhyk-1)? without the description text?
> 
Fixed 
Screenshot 2024-09-09 at 1 23 05 PM



> > Shows an alert instead of 404 error page when the user is not authorized to access the plugin (https://issues.redhat.com/browse/RHIDP-3833)
> 
> I guess this is fine for now, but it will be great using the [full page layout] (https://www.figma.com/design/GeVb7rc390TkxeQNnJIcBD/RHDH-Theme-(pages)?node-id=110-933&t=tedPPRSTpsge64S3-1) for this use cases in the future (probably we need another ticket).
> 

I created this ticket https://issues.redhat.com/browse/RHIDP-3889 , to address the above

**Comment by invincibleJai** [comment]
Thanks @debsmita1, i have one observation

The icon and text are not aligned properly

image

**Comment by debsmita1** [comment]
> Thanks @debsmita1, i have one observation
> 
> The icon and text are not aligned properly
> 
> image

Fixed
Screenshot 2024-09-09 at 4 43 55 PM

**Comment by invincibleJai** [comment]
getting , here there is error + some data

image

**Comment by rm3l** [comment]
> getting below, here there is error + some data
> 
> image

Indeed, errors returned by GitHub do not prevent the backend from fetching the rest of the data. To help with troubleshooting, those errors are collected and returned in the `errors` field. So depending on the permissions of the GH integrations, some data might be returned, but errors might be returned as well.
IMO, we should still display the data returned along with any potential errors from the `errors` field.

**Review by rm3l — COMMENTED** [review]
Tested this PR, and it was working as expected. LGTM, but I'll let UI folks take another look.
Thanks.

File: plugins/bulk-import/src/components/PreviewFile/PreviewFile.tsx [hunk 1]
```diff
@@ -80,55 +90,111 @@ export const PreviewFileSidebar = ({
...
+  const initializePullRequest = React.useCallback(async () => {
     const newPullRequestData: PullRequestPreviewData = {};
 
     if (Object.keys(data?.selectedRepositories || [])?.length > 0) {
-      Object.values(data?.selectedRepositories || []).forEach(repo => {
-        if (repo.catalogInfoYaml?.prTemplate) {
-          newPullRequestData[repo.id] = repo.catalogInfoYaml.prTemplate;
+      setIsLoading(true);
+      Object.values(data?.selectedRepositories || []).forEach(async repo => {
+        const result = await bulkImportApi.getImportAction(
```
Comment (invincibleJai) [inline plugins/bulk-import/src/components/PreviewFile/PreviewFile.tsx:119]: async await might not work as expected with forEach

https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop

File: plugins/bulk-import/src/components/PreviewFile/PreviewPullRequestForm.tsx [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+  },
+  previewCardContent: {
+    paddingTop: 0,
+  },
+}));
+
+const componentNameRegex =
+  /^([a-zA-Z0-9]+[-_.])*[a-zA-Z0-9]+$|^[a-zA-Z0-9]{1,63}$/;
+
+export const PreviewPullRequestForm = ({
```
Comment (invincibleJai) [inline plugins/bulk-import/src/components/PreviewFile/PreviewPullRequestForm.tsx:47]: good to have test for component

File: plugins/bulk-import/src/components/PreviewFile/PreviewPullRequestForm.tsx [hunk 3]
```diff
@@ -0,0 +1,495 @@
...
+  previewCard: {
+    marginTop: theme.spacing(1),
+  },
+  previewCardContent: {
+    paddingTop: 0,
+  },
+}));
+
+const componentNameRegex =
+  /^([a-zA-Z0-9]+[-_.])*[a-zA-Z0-9]+$|^[a-zA-Z0-9]{1,63}$/;
```
Comment (invincibleJai) [inline plugins/bulk-import/src/components/PreviewFile/PreviewPullRequestForm.tsx:45]: can we move regex and it's usage to util and add test for it?

File: plugins/bulk-import/src/hooks/useAddedRepositories.ts [hunk 4]
```diff
@@ -21,11 +25,17 @@ export const useAddedRepositories = (
...
+  const configApi = useApi(configApiRef);
   const { value: user } = useAsync(async () => {
     const identityRef = await identityApi.getBackstageIdentity();
     return identityRef.userEntityRef;
   });
 
+  const { value: baseUrl } = useAsync(async () => {
+    const url = configApi.getString('app.baseUrl');
+    return url;
+  });
```
Comment (invincibleJai) [inline plugins/bulk-import/src/hooks/useAddedRepositories.ts:37]: nit: this is duplicated below as well, good to have it in one place?

File: plugins/shared-react/src/components/common/DrawerContext.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+};
+
+export const DrawerContext = createContext<DrawerContextType>({
+  drawerData: {},
+  setDrawerData: () => {},
+  openDrawer: false,
+  setOpenDrawer: () => {},
+});
+
+export const DrawerContextProvider = (props: any) => {
```
Comment (invincibleJai) [inline plugins/shared-react/src/components/common/DrawerContext.tsx:17]: nit: type for props?

File: plugins/bulk-import/src/components/PreviewFile/PreviewFile.tsx (hunk 1 above)
Comment (debsmita1) [reply plugins/bulk-import/src/components/PreviewFile/PreviewFile.tsx]: You are right, this was overlooked. Thank you

Fixed it now

**Comment by debsmita1** [comment]
> getting below, here there is error + some data
> 
> image

Fixed it. 
Showing the error along with the data that is fetched

**Review by invincibleJai — APPROVED** [review]
/lgtm

**Pull request merged**